src/sentry/static/sentry/app/components/contextPickerModal.jsx

```diff
@@ -0,0 +1,255 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import {fetchOrganizationDetails} from '../actionCreators/organizations';
+import {t} from '../locale';
+import LatestContextStore from '../stores/latestContextStore';
+import LoadingIndicator from './loadingIndicator';
+import OrganizationsStore from '../stores/organizationsStore';
+import Select2Field from './forms/select2Field';
+import SentryTypes from '../proptypes';
+import replaceRouterParams from '../utils/replaceRouterParams';
+import withProjects from '../utils/withProjects';
+
+class ContextPickerModal extends React.Component {
+  static propTypes = {
+    /**
+     * The destination route
+     */
+    nextPath: PropTypes.string.isRequired,
+
+    /**
+     * Finish callback
+     */
+    onFinish: PropTypes.func.isRequired,
+
+    /**
+     * Container for modal header
+     */
+    Header: PropTypes.oneOfType([PropTypes.element, PropTypes.func, PropTypes.string]),
+
+    /**
+     * Container for modal body
+     */
+    Body: PropTypes.oneOfType([PropTypes.element, PropTypes.func, PropTypes.string]),
+
+    /**
+     * List of available organizations
+     */
+    organizations: PropTypes.arrayOf(SentryTypes.Organization),
+
+    /**
+     * LatestContext store
+     */
+    latestContext: PropTypes.shape({
+      organization: SentryTypes.Organization,
+    }),
+
+    /**
+     * Does modal need to prompt for organization.
+     * TODO(billy): This can be derived from `nextPath`
+     */
+    needOrg: PropTypes.bool,
+
+    /**
+     * Does modal need to prompt for project
+     */
+    needProject: PropTypes.bool,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: props.organizations.length === 1,
```

Comment:
I got confused by this. But after reading, I get it. This component only shows the loading animation **if** it has successfully decided on an organization to redirect to. Right? Could we have a comment that reflects that?